File: leet_387_first_uniq_char.cpp
class Solution {
public:
    int firstUniqChar(string s)
    {
        vector<int> E(26, 0);
        int         i = -1;
        while (++ i < s.length())
            E[s[i] - 'a'] ++ ;
        i = -1;
        while (++ i < s.length())
            if (E[s[i] - 'a'] == 1) return i ;
        return -1 ;
    }
};

File: leet_387_first_uniq_char.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        E = [0] * 26;
        for c in s:
            E[ ord(c) - ord('a') ] += 1
        i = -1
        for c in s:
            i += 1
            if E[ ord(c) - ord('a') ] == 1:
                return i
        return -1

    
